src/back/auth/src/service/tokens_oidc.cpp

```diff
@@ -25,6 +29,7 @@ struct jwt_impl {
 
 	Verifier* GetVerifier(const std::string& keyId, const std::string& algoName)
 	{
+		std::lock_guard<engine::Mutex> lock(mutex);
 		auto it = _verifier.find(keyId);
 		if (it != _verifier.end())
 			return it->second.get();
```

Comment:
Нужно использовать shared mutex потому что у нас много операций на чтение и мало на запись.

Для вызова только на чтение нужно использовать shared_lock а на запись обычный блокирующий чтение.
```suggestion
		{
			std::shared_lock lock(_mutex);
			auto it = _verifier.find(keyId);
			if (it != _verifier.end())
				return it->second.get();
		}

		std::lock_guard lock(_mutex);
```

Reply:
Готово.